lib/semantics/resolve-names.cc

```diff
@@ -1099,7 +1207,62 @@ class OmpVisitor : public virtual DeclarationVisitor {
     return false;
   }
 
+  void Post(const parser::Name &);
+
 private:
+  struct OmpContext {
+    OmpContext(const parser::CharBlock &source, OmpDirective d)
+      : directiveSource{source}, directive{d} {}
+    parser::CharBlock directiveSource{nullptr};
+    OmpDirective directive;
+    Scope *scope{nullptr};
+    // TODO: default DSA is implicitly determined in different ways
+    Symbol::Flag defaultDSA{Symbol::Flag::OmpShared};
+    // variables on Data-sharing attribute clauses
+    std::map<const Symbol *, Symbol::Flag> objectMap;
```

Comment:
The name `objectMap` doesn't say much about what it is for. How about `dataSharingAttributes` or `objectToDSA`? Then rename `SetContextObjectMap` and `HasObject` to reflect the new name.

Reply:
I renamed it to `objectWithDSA`.